tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go

```diff
@@ -580,7 +577,38 @@ func (a *AwsProvider) addPortToSg(sgId string, port int64) error {
 	return nil
 }
 
-// findWindowsWorkerSg creates the Windows worker security group with name <infraID>-windows-worker-sg.
+// createNewRulesForSg creates a new list of rules to be added in newly created sg
+func createNewRulesForSg(vpc *ec2.Vpc) ([]*ec2.IpPermission, error) {
+	myIP, err := GetMyIp()
+	if err != nil {
+		return nil, fmt.Errorf("error getting IP: %s", err)
+	}
+
+	populatedRules := make([]*ec2.IpPermission, 0)
+	ports := []int64{WINRM_PORT, sshPort, rdpPort}
+
+	for _, port := range ports {
+		populatedRules = append(populatedRules,
+			(&ec2.IpPermission{}).
```

Comment:
This is going to attempt to add this rule multiple times, you should remove it from the for loop